test/validators/properties-validator.test.ts

```diff
@@ -1,121 +1,263 @@
-import { PropertiesValidator } from '../../lib/validators/properties-validator';
+import { PropertiesValidator } from '../../lib/validators';
+import type { ComponentProperty } from '../../lib/models';
+import cloneDeep = require('lodash.clonedeep');
 
 describe('PropertiesValidator', () => {
-    let definition: any;
-    let error: jasmine.Spy;
-    let validator: PropertiesValidator;
-    let fileList: Set<string>;
-    beforeEach(() => {
-        // valid definition (cut)
-        definition = {
-            version: '1.0.0',
+    const textProperty: ComponentProperty = {
+        name: 'textProperty',
+        control: {
+            type: 'text',
+        },
+    } as any;
+    const radioProperty: ComponentProperty = {
+        name: 'radioProperty',
+        control: {
+            type: 'radio',
+            options: [{ icon: 'path1' }, { icon: 'path2' }, { icon: 'path5' }],
+        },
+    } as any;
+    const mediaProperty: ComponentProperty = {
+        name: 'mediaProperty',
+        control: {
+            type: 'media-properties',
+        },
+        dataType: 'doc-media',
+    } as any;
+    const headerProperty: ComponentProperty = {
+        control: {
+            type: 'header',
+        },
+        label: 'header-label',
+    } as any;
+
+    function createConditionalProperty(children: any[]) {
+        return {
+            name: 'conditionalProperty',
+            control: {
+                type: 'text',
+            },
+            childProperties: [
```

Comment:
This is not the right construction, right? I would expect it has the matchExpression structure?

Reply:
For these tests the matchExpression is not needed, but it now causes confusion

Reply:
Added it